python/lsst/analysis/ap/nb_utils.py

```diff
@@ -70,3 +78,203 @@ def make_simbad_link(ra, dec, radius_arcsec=3.0):
+    We could relax that assumption later, though it would require
+    removing the visit, detector loop.
+    """
+
+    visit_det = set(zip(goodSrc1['visit'], goodSrc1['detector']))
+    src1 = []
+    src2 = []
+    matches = []
+    indices = {}
+    seps = {}
+
+    for visit, detector in visit_det:
+        mask1 = (goodSrc1['visit'] == visit) &\
+            (goodSrc1['detector'] == detector)
+        mask2 = (goodSrc2['visit'] == visit) &\
+            (goodSrc2['detector'] == detector)
+        gs1 = goodSrc1[mask1].copy()
+        gs2 = goodSrc2[mask2].copy()
+
+        coords1 = SkyCoord(ra=gs1['ra'].values*u.degree,
+                           dec=gs1['decl'].values*u.degree)
+        coords2 = SkyCoord(ra=gs2['ra'].values*u.degree,
+                           dec=gs2['decl'].values*u.degree)
+
+        index, sep, _ = match_coordinates_sky(coords1, coords2)
+        indices[(visit, detector)] = index
+        seps[(visit, detector)] = sep
+        gs1['xmatch_dist_arcsec'] = sep.to_value(u.arcsecond)
+        gs1['src2_diaSourceId'] = gs2['diaSourceId'].values.astype(np.int64)[index]
+
+        # Set the match ID to 0 if the distance is above threshold
+        gs1.loc[(gs1['xmatch_dist_arcsec'] > matchRadius),
+                ['src2_diaSourceId']] = 0
+
+        # get the DiaSources in dataset 2 not matched to something in dataset 1
+        uniqueid2 = set(gs2['diaSourceId']) - set(gs1['src2_diaSourceId'])
+        unique2 = gs2[gs2['diaSourceId'].isin(uniqueid2)]
+
+        unique1 = gs1[(gs1['src2_diaSourceId'] == 0)]
+
+        withmatch = gs1[(gs1['src2_diaSourceId'] > 0)]
+
+        src1.append(unique1)
+        src2.append(unique2)
+        matches.append(withmatch)
+
+    # Out of the loop, concatenate everything together.
+    unique1 = pd.concat(src1)
+    unique2 = pd.concat(src2)
+    matched = pd.concat(matches)
+
+    # Make cutouts if they don't already exist; optionally display to screen
+    config = pisco.PlotImageSubtractionCutoutsConfig()
+    config.url_root = ''
+    config.add_metadata = add_cutout_metadata
+    config.diff_image_type = diffimg_type1
+
+    if cutoutPath is None:
+        cutoutPath = os.environ['PWD']
+        logging.warning('cutoutPath not supplied; defaulting to '+cutoutPath)
+
+    piscot = pisco.PlotImageSubtractionCutoutsTask(output_path=cutoutPath,
+                                                   config=config)
+
+    print("{} matched sources; {} unique to set 1; {} unique to set 2.".format(
+        len(matched), len(unique1), len(unique2)))
+
+    # First figure out which cutouts already exist at the output path
+    unique1['pathexists'] = False
+    for i in range(len(unique1)):
+        dId = unique1.iloc[i]['diaSourceId']
+        idx = unique1.index[i]
+        unique1.at[idx, 'pathexists'] = os.path.exists(
+            os.path.join(
+                cutoutPath,
+                f"images/{int(dId // chunk_size) * chunk_size}/{dId}.png"))
```

Comment:
I suggest removing the chunking altogether. The diaSource Ids are not continuous, so even with the large default size of 10,000 I still get 5 subdirectories of cutouts, even with only a few dozen cutouts.

Reply:
Also, it would be very helpful to have the cutouts that are unique to dataset 1 in a different sub-directory than the cutouts that are unique to dataset 2.

Reply:
I was thinking to split into different subdirectories as well.

Reply:
Please use `plotImageSubtractionCutouts.CutoutPath` when dealing with paths to cutout files instead of managing the paths yourself.

Reply:
Will follow John's suggestion on using the CutoutPath as well as Ian's to offer separate paths for each set of cutouts.

Reply:
What I converged on was making the chunk size for each dataset its own optional parameter, defaulting to None. That way if you did happen to have a pre-existing set of cutouts for one or both of the datasets you could still use it (as long as you were careful to set the chunk size to match what was used for creation) and it would detect they were already there. On the other hand, if you didn't have a huge number of sources and you didn't want to mess with the various subdirectories, then you could just omit the option and it would skip it.